src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/BulkNumberSearchParameterV2RowGenerator.cs

```diff
@@ -0,0 +1,32 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal class BulkNumberSearchParameterV2RowGenerator : BulkSearchParameterRowGenerator<NumberSearchValue, BulkNumberSearchParamTableTypeV2Row>
+    {
+        public BulkNumberSearchParameterV2RowGenerator(SqlServerFhirModel model, SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+            : base(model, searchParameterTypeMap)
+        {
+        }
+
+        internal override bool TryGenerateRow(int offset, short searchParamId, NumberSearchValue searchValue, out BulkNumberSearchParamTableTypeV2Row row)
+        {
+            var singleValue = searchValue.Low == searchValue.High ? searchValue.Low : null;
+
+            row = new BulkNumberSearchParamTableTypeV2Row(
+                offset,
+                searchParamId,
+                singleValue.HasValue ? singleValue : null,
+                (decimal)(singleValue.HasValue ? singleValue : searchValue.Low ?? (decimal?)VLatest.NumberSearchParam.LowValue.MinValue),
```

Comment:
>VLatest.NumberSearchParam.LowValue.MinValue [](http://example.com/codeflow?start=92&length=43)

I know you didn't wrote this part, but what condition we use during search to filter things out?

let's say we have two records and each has range (null, 100) and (decimal.MinValue, 100)

I do search saying I'm looking for range which starts with decimal.MinValue.

Are we gonna return first record or not with current changes?

Reply:
We will return both records since if one of the value low/high is not provided in the range, that is translated to minValue/maxValue in the NumberSearchParam table and this holds true without current changes as well.